examples/server/server.cpp

```diff
@@ -605,7 +610,9 @@ struct server_task_result_cmpl_final : server_task_result {
 
 struct server_task_result_cmpl_partial : server_task_result {
     int index = 0;
-    std::string content;
+
+    std::string  content;
+    llama_tokens tokens;
```

Comment:
P/s: we cannot `std::move` it, because inside `send_partial_response`, `result` is still being used after `send_partial_response`

Reply:
I'm not really sure that the `"return_tokens"` logic is necessary. The `tokens` array should be similar in JSON length to the `content` string, though I am not sure performance wise how much slower it is to serialize an array of integers compared to a string. Anyway, I've added the flag and added tests.

Note that with `"stream": true` we always return the `tokens` field in the partial responses (i.e. this is not affected by the `"return_tokens"` flag).

Reply:
What I'm thinking is that this should not degrade the performance of JSON serializing/parsing. But I'm just thinking about the bandwidth, because it seems like in most cases, we're now using double the bandwidth.

For `stream`, I don't think it's a problem because time to serialize/send/receive/parse is minor compared to the time a token is generated.

But I think for now we can keep it this way. The non-OAI `/completion` is a playground anw so it's fine to expose everything. The `/v1/completions` that I'm planning to do next will be more prod-ready, thus it won't have these data in the response.

Reply:
> But I think for now we can keep it this way. The non-OAI /completion is a playground anw so it's fine to expose everything. The OAI compat /v1/completions that I'm planning to do next will be more prod-ready, thus it won't have these data in the response.

Yes, I agree we can keep `/v1/completions` strongly OAI-compat (i.e. not even have extra fields like `tokens`) and only have these in the non-OAI endpoints like `/completions`.

Reply:
> you implemented `return_tokens`, that's good then, let's keep it 👍

This is great to see, thank you.

I sometimes use the `/completion` API on a bandwidth-constrained network (Wireguard over a bad WAN connection) so having an option to disable tokens if I don't need them is perfect.